### PR TITLE
ISPN-15425 TopologyChangeFunctionalTest random failures

### PR DESCRIPTION
**Pull request by pruivo**
https://issues.redhat.com/browse/ISPN-15425

Unfortunately, it does not fix all flaky tests but should make them rare :)

### CODE REVIEW

**Comment by pruivo** [comment]
follow up: https://issues.redhat.com/browse/ISPN-15426

**Comment by jabolina** [comment]
CI timed out. I replayed it.

**Review by jabolina — APPROVED** [review]

File: server/hotrod/src/test/java/org/infinispan/server/hotrod/tx/TopologyChangeFunctionalTest.java [hunk 1]
```diff
@@ -114,10 +119,13 @@ public void testNodeJoin(Method method) {
 
    @Test(groups = "unstable", description = "ISPN-8432")
    public void testOriginatorLeft(Method method) {
-      final byte[] k1 = k(method, "k1");
-      final byte[] k2 = k(method, "k2");
-      final byte[] v1 = v(method, "v1");
-      final byte[] v2 = v(method, "v2");
+      //optimistic mode is not official supported and we don't store any versioning
+      //this test is failing without the versions
+      SkipTestNG.skipIf(lockingMode == LockingMode.OPTIMISTIC, "Optimistic transactions not supported");
```
[the remark this answers is not in the document]
Comment (pruivo) [reply server/hotrod/src/test/java/org/infinispan/server/hotrod/tx/TopologyChangeFunctionalTest.java:124]: yes, it is failing randomly quite often :rofl:

**Comment by jabolina** [comment]
The `InvalidServerConfigTxTest.testOkConfig` seems to always fail. Is it related?

**Comment by pruivo** [comment]
@jabolina yes. it is a `NullPointerException`. Should be fixed now.

**Comment by jabolina** [comment]
Thanks, @pruivo!

**Pull request merged**